pages/settings/account/index.tsx

```diff
@@ -268,10 +270,14 @@ const AccountSettings = () => {
     }
   })
 
-  const loading = userInfoLoading || status === 'loading'
+  const loading = userInfoLoading || status === SessionStatus.Loading
+  const unauthenticated = status === SessionStatus.unauthenticated
   const { username, name, discordUsername } = data?.userInfo?.user || {}
 
-  if (loading || !data) {
```

Comment:
what happens if no data is returned from the userinfoquery? or if there is an error in making the request?

Reply:
If there's no data it means the user is unauthenticated and they'll be redirected to `/login` as we discussed above. It doesn't handle errors. We can make it handle errors in a separate PR.

Reply:
My concern is that in your PR description it says `After this PR is merged, we'll release the page and share it with the students`. If features are getting released into production, error scenarios need to be accounted for as a requirement. We cannot have an incomplete experience for any scenario, even during error scenarios. All edge cases should be taken care of.

If we can merge the changes in this PR without releasing the changes to production, then it is okay to make a followup PR to account for error scenarios. Otherwise, they need to be addressed in this PR.

Reply:
You got a point. We'll just make this PR about creating the hook, and after resolving the issues mentioned in this PR, we can then release it.